src/main/kotlin/com/tripl3dogdare/havenjson/Utils.kt

```diff
@@ -1,33 +1,31 @@
 package com.tripl3dogdare.havenjson
 
-internal tailrec fun String.jsonEscape(acc:String=""):String {
-  if(this.isEmpty()) return acc
-  return when(this[0]) {
-    '\"' -> tail.jsonEscape("$acc\\\"")
-    '\\' -> tail.jsonEscape("$acc\\\\")
-    '\n' -> tail.jsonEscape("$acc\\n")
-    '\b' -> tail.jsonEscape("$acc\\b")
-    '\r' -> tail.jsonEscape("$acc\\r")
-    '\t' -> tail.jsonEscape("$acc\\t")
-    '\u000C' -> tail.jsonEscape("$acc\\f")
-    else -> tail.jsonEscape(acc+this[0])
+internal fun String.jsonEscape() = fold("") { acc, c ->
+  acc + when (c) {
+    '\"' -> """\""""
+    '\\' -> """\\"""
+    '\n' -> """\n"""
+    '\b' -> """\b"""
+    '\r' -> """\r"""
+    '\t' -> """\t"""
+    '\u000C' -> """\f"""
+    else -> c
   }
 }
 
-internal tailrec fun String.jsonUnescape(acc:String=""):String {
-  if(this.isEmpty()) return acc
-  return if(this[0] == '\\') when(this[1]) {
-    '"' -> drop(2).jsonUnescape("$acc\"")
-    '\\' -> drop(2).jsonUnescape("$acc\\")
-    'n' -> drop(2).jsonUnescape("$acc\n")
-    'b' -> drop(2).jsonUnescape("$acc\b")
-    'r' -> drop(2).jsonUnescape("$acc\r")
-    't' -> drop(2).jsonUnescape("$acc\t")
-    'f' -> drop(2).jsonUnescape("$acc\u000c")
-    'u' -> drop(6).jsonUnescape(acc+this.drop(2).take(4).toInt(16).toChar())
-    else -> drop(2).jsonUnescape(acc+this[1])
-  } else tail.jsonUnescape(acc+this[0])
-}
+internal fun String.jsonUnescape() = replaceAll(mapOf(
```

Comment:
I'm curious why this particular method with the `replaceAll` method instead of using `fold` directly similar to `jsonEscape`. Is there any particular reason? If not, I'd prefer to see it written in a similar fashion, primarily for ease of understanding when comparing the two.

Reply:
`fold` can't be used in `jsonUnescape` because, unlike `jsonEscape` where you iterate over the string character by character, here you have to look ahead a character or more. You can use `foldIndexed` to get a substring every time you find a `\\` but that defeats the purpose of using `fold`.
On another note, `replaceAll` is not needed after all because `jsonUnescape` can be done with just two `replace`s with regexes. The following commit will fix this and remove `replaceAll` (if you want to keep it, you can copy it over from the first commit and make it `internal`)

Reply:
Sounds good!